src/mesh/mesh.cc

```diff
@@ -219,15 +221,18 @@ void Mesh::init_from_input() {
     Input::Array region_list;
     RegionDB::MapElementIDToRegionID el_to_reg_map;
 
-    // create regions from our input
-    if (in_record_.opt_val("regions", region_list)) {
-        region_db_.read_regions_from_input(region_list, el_to_reg_map);
-    }
     // read raw mesh, add regions from GMSH file
     GmshMeshReader reader( in_record_.val<FilePath>("mesh_file") );
     reader.read_mesh(this, &el_to_reg_map);
```

Comment:
We can't remove this code, it relates with key 'element_list', that is not removed or changed (maybe some changes will be in next issue #386). Marked 'if (in_record_.opt_val("regions" ...' is not removed, only moved after call setup_topology.